tests/conftest.py

```diff
@@ -1379,24 +1379,43 @@ def collect_db_dump_on_duts(request, duthosts):
     if hasattr(request.node, 'rep_call') and request.node.rep_call.failed:
         dut_file_path = "/tmp/db_dump"
         docker_file_path = "./logs/db_dump"
-        db_dump_path = os.path.join(dut_file_path, request.module.__name__, request.node.name)
         db_dump_tarfile = "{}.tar.gz".format(dut_file_path)
 
         # Collect DB config
         dbs = set()
         result = duthosts[0].shell("cat /var/run/redis/sonic-db/database_config.json")
         db_config = json.loads(result['stdout'])
+        state_db_id = db_config['DATABASES']['STATE_DB']['id']
         for db in db_config['DATABASES']:
             db_id = db_config['DATABASES'][db]['id']
+            # Skip STATE_DB dump on release 201911.
+            # JINJA2_CACHE can't be dumped by "redis-dump", and it is stored in STATE_DB on 201911 release.
+            # Please refer to issue: https://github.com/Azure/sonic-buildimage/issues/5587.
+            # The issue has been fixed in https://github.com/Azure/sonic-buildimage/pull/5646.
+            # However, the fix is not included in 201911 release. So we have to skip STATE_DB on release 201911
+            # to avoid raising exception when dumping the STATE_DB.
+            if i == state_db_id and duthosts[0].sonic_release in ['201911']:
```

Comment:
@xwjiang2021 This  line is giving a error as 'i' as undefined. Should this be db_id instead of i